moveit_planners/pilz_industrial_motion_planner/src/trajectory_generator_circ.cpp

```diff
@@ -142,54 +160,85 @@ void TrajectoryGeneratorCIRC::extractMotionPlanInfo(const planning_scene::Planni
     {
       frame_id = req.goal_constraints.front().position_constraints.front().header.frame_id;
     }
-    info.goal_pose = getConstraintPose(req.goal_constraints.front());
-  }
 
-  assert(req.start_state.joint_state.name.size() == req.start_state.joint_state.position.size());
-  for (const auto& joint_name : robot_model_->getJointModelGroup(req.group_name)->getActiveJointModelNames())
-  {
-    auto it{ std::find(req.start_state.joint_state.name.cbegin(), req.start_state.joint_state.name.cend(), joint_name) };
-    if (it == req.start_state.joint_state.name.cend())
+    const auto link = robot_model_->getLinkModel(frame_id);
+    if (link)
     {
+      // check if robot link is rigidly connected to the root link
+      const auto parent_link = robot_model_->getRigidlyConnectedParentLinkModel(link);
```

Comment:
There is also:
https://github.com/ros-planning/moveit/blob/167a62176b69a3afe102a2c9696aaf20225c52e4/moveit_core/robot_state/include/moveit/robot_state/robot_state.h#L1389-L1390

which allows to combine these two cases.